virtwho/virt/rhevm/rhevm.py

```diff
@@ -71,18 +71,73 @@ def __init__(self, logger, config):
         self.username = self.config.username
         self.password = self.config.password
 
-        self.clusters_url = urlparse.urljoin(self.url, "api/clusters")
-        self.hosts_url = urlparse.urljoin(self.url, "api/hosts")
-        self.vms_url = urlparse.urljoin(self.url, "api/vms")
-
         self.auth = HTTPBasicAuth(self.config.username, self.config.password)
 
+    @property
+    def clusters_url(self):
+        if not hasattr(self, '_clusters_url'):
+            if self.major_version == '4' and 'ovirt-engine' not in self.url:
```

Comment:
Could we do the version check on init and update self.url to include the necessary prefix (for the old rhev set prefix to 'api/', for new set prefix to 'ovirt-engine/api/') ?